app/src/main/java/com/andrew/tasky/auth/data/AuthRepositoryImpl.kt

```diff
@@ -85,4 +86,14 @@ class AuthRepositoryImpl(
         prefs.clearPrefs()
         agendaRepository.deleteAllAgendaTables()
     }
+
+    override suspend fun isAuthorizedToLogin(): Boolean {
+        val result = authenticate()
+        when (result) {
+            is Resource.Error -> {
+                return result.errorType is ApiErrorType.IOException && prefs.containsJwt()
```

Comment:
Actually, here you only want to check for HTTP 401 specifically which means not authorized. For other HTTP status codes such as 500 (when there's an error server-side which is not the client's fault), you don't want to treat this as unauthorized